### PR TITLE
[HACKATHON] Email notification option on gem2s loading screen 

### PR DESCRIPTION
**Pull request by StefanBabukov**
# Background
#### Link to staging deployment URL 
https://ui-stefan-ui532-api241.scp-staging.biomage.net/
#### Links to any Pull Requests related to this

UI - https://github.com/biomage-ltd/ui/pull/532
API - https://github.com/biomage-ltd/api/pull/241
pipeline - https://github.com/biomage-ltd/pipeline/pull/180

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by aerlaut** [comment]
So the pipeline status notification is not enabled by default? I thought there was discussion to have this enabled by default and the option to change this settings moved under the user settings menu?

File: src/__test__/components/NotifyByEmail.test.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
```
Comment (aerlaut) [inline src/__test__/components/NotifyByEmail.test.jsx:1]: Can you change this experiment to use the new integration test? As an example, you can follow how it is structured at the `markerHeatmap.test.js`

File: src/__test__/redux/reducers/experimentsReducer.test.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ describe('experimentsReducer', () => {
       type: '10x',
     },
     sampleIds: [],
+    notifyByEmail: false,
```
Comment (aerlaut) [inline src/__test__/redux/reducers/experimentsReducer.test.js]: Does this mean the option to get notified depends on the experiment? not the user?

Comment (StefanBabukov) [reply src/__test__/redux/reducers/experimentsReducer.test.js]: yeah its on the experiment. I think if we want it to depend on the user,  we will need to add a new field in cognito? After that migrate all the users so they have the field. I don't know if we should do that?

File: src/components/NotifyByEmail.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    if (!activeProjectUuid) {
+      dispatch(loadProjects());
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!experiments.ids?.length && activeProjectUuid) {
+      dispatch(loadExperiments(activeProjectUuid));
+    }
+  }, [activeProjectUuid]);
```
Comment (aerlaut) [inline src/components/NotifyByEmail.jsx]: Do you still need this?

Comment (StefanBabukov) [reply src/components/NotifyByEmail.jsx]: I need this in the case the user reloads the page and the redux store doesn't have the experiment loaded in (I use the field "notifyByEmail" there)

File: src/__test__/redux/reducers/experimentsReducer.test.js (hunk 2 above)
Comment (aerlaut) [reply src/__test__/redux/reducers/experimentsReducer.test.js]: I see.. ok. Have you asked Vicky or Marcell the UX side of this?

Comment (StefanBabukov) [reply src/__test__/redux/reducers/experimentsReducer.test.js]: Vicky said she is happy with the UX side

File: src/__test__/components/NotifyByEmail.test.jsx [hunk 4]
```diff
@@ -3,96 +3,73 @@ import {
...
   };
+  it('loads experiments if  there is an activeProjectUuid', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
+    await renderNotifyByEmail();
+    expect(loadExperimentsSpy).toHaveBeenCalledTimes(1);
+  });
 
   it('Renders Correctly', () => {
-    renderNotifyByEmail(defaultState);
+    renderNotifyByEmail();
```
Comment (aerlaut) [inline src/__test__/components/NotifyByEmail.test.jsx]: I would `await` here to make sure that `renderNotifyByEmail` is resolved before the next LOC Is executed

**Review by aerlaut — APPROVED** [review]
UI side looks good 👍

**Pull request merged**